src/Microsoft.VisualStudio.Editors/PropPages/ApplicationPropPageInternalBase.vb

```diff
@@ -135,14 +135,18 @@ Namespace Microsoft.VisualStudio.Editors.PropertyPages
                     Dim supportedFrameworks As IEnumerable(Of TargetFrameworkMoniker) = TargetFrameworkMoniker.GetSupportedTargetFrameworkMonikers(vsFrameworkMultiTargeting, DTEProject, supportedTargetFrameworksDescriptor?.Converter)
```

Comment:
`GetSupportedTargetFrameworkMonikers` returns the result of `GetSupportedTargetFrameworks`

It seems all implementation of `GetSupportedTargetFrameworks` can be changed to return a list (the implementations calls `ToList` internally already). Then, `GetSupportedTargetFrameworkMonikers` can return a list as well.

After that, The code can just say:

```
If supportedFrameworks.Count > 0 Then
    targetFrameworkComboBox.Items.AddRange(supportedFrameworks)
Else
    targetFrameworkSupported = False
    targetFrameworkComboBox.Items.Clear()
End If
```

Reply:
Good point. I've changed those methods to return IROL and reverted my changes here.